src/main/java/com/algorand/algosdk/builder/transaction/MethodCallTransactionBuilder.java

```diff
@@ -0,0 +1,171 @@
+package com.algorand.algosdk.builder.transaction;
+
+import com.algorand.algosdk.abi.Method;
+import com.algorand.algosdk.crypto.Address;
+import com.algorand.algosdk.crypto.TEALProgram;
+import com.algorand.algosdk.logic.StateSchema;
+import com.algorand.algosdk.transaction.MethodCallParams;
+import com.algorand.algosdk.transaction.Transaction;
+import com.algorand.algosdk.transaction.TxnSigner;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+@SuppressWarnings("unchecked")
+public class MethodCallTransactionBuilder<T extends MethodCallTransactionBuilder<T>> extends TransactionParametersBuilder<T> implements StateSchemaSetter<T>, TEALProgramSetter<T>, ApplicationCallReferencesSetter<T> {
```

Comment:
@jasonpaulos I opened an optional child PR for your consideration:  https://github.com/algorand/java-algorand-sdk/pull/326.

Unless I misunderstand usage, I _think_ it's safe to remove the generic type for non-abstract members.  The PR illustrates the change.

Though as I'm writing, I see the other pre-existing, non-abstract classes still expose `T` (e.g. `AssetTransferTransactionBuilder`).  That's different than what I expected.  Since I already generated the PR, I'll leave as is for feedback.

Reply:
The delegation vs inheritance change from that PR looks good -- I don't have a strong opinion here, but if delegation is preferred, that will solve the problem just as well.

As for the templating issue, the only technical drawback I'm aware of for removing the template is that `MethodCallTransactionBuilder` can't be subclassed. While that doesn't matter much currently, I'd rather not prohibit us from a subclass in the future, especially since all the other builders follow this same pattern.

Reply:
@jasonpaulos Alright - I'll close #326.  Resolving.

I think subclassing a concrete type is _dangerous_.  However, that's the repo's convention, so let's keep as is.

Reply:
I don't remember the details, but I spent a lot of time back then messing around with different syntax to make sure the IDE was able to figure out exactly what options were valid.

i.e. when it wasn't working, if you were building an app call transaction and set a common field like "fee", autocomplete would no longer suggest "applicationID".

Reply:
Aside from that, I completely agree that this is pretty weird. If there is something simpler that would be nice.

Reply:
We could get rid of the unchecked operations warnings with an approach like this: https://stackoverflow.com/a/5818701

But it doesn't change the templating syntax at all.

I'm generally in favor of a change like this, but we'd probably want to do it in a new PR, as the changes would be extensive.

Reply:
@jasonpaulos I agree, thats a great solution.